gov2/demotools/question.go

```diff
@@ -0,0 +1,149 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package demotools provides a set of tools that help you write code examples.
+//
+// **Questioner**
+//
+// The questioner is used in interactive examples to ask for input from the user
+// at a command prompt, validate the answer, and re-ask the question if needed.
+// It is exposed through an interface so that it can be mocked for unit testing.
+// A pre-written mock is provided in the testtools package.
+package demotools
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// IAnswerValidator defines an interface that validates string input.
+type IAnswerValidator interface {
+	Validate(string) bool
```

Comment:
consider making this `isValid(string) bool` so subsequent calls read nicer e.g. `if validator.isValid(answer) {`